_pytest/logging.py

```diff
@@ -368,12 +371,33 @@ class _LiveLoggingStreamHandler(logging.StreamHandler):
     """
     Custom StreamHandler used by the live logging feature: it will write a newline before the first log message
     in each test.
+
+    During live logging we must also explicitly disable stdout/stderr capturing otherwise it will get captured
```

Comment:
I guess this is the reason that when I enabled log_cli I can't get any logs into the junitxml file. Junitxml plugin AFAIK gets all captured logs sections and puts them inside the xml file. If you don't capture logs here, when the log_cli is enabled, we will never be able to see logs in console and in the xml file.

Reply:
Good point @KKoukiou.

Looking at your configuration:

```yaml
version: 1
handlers:
  consoleHandler:
    class: logging.StreamHandler
    level: INFO
    stream: ext://sys.stdout
  simpleFileHandler:
    class: logging.FileHandler
    filename: logs.txt
loggers:
  art.flow:
    level: INFO
    handlers:
      - consoleHandler
    propagate: 0
root:
  level: DEBUG
  handlers:
    - simpleFileHandler
disable_existing_loggers: False
```

I believe if you configure the `root` logger to also write to `stdout`:

```yaml
root:
  level: DEBUG
  handlers:
    - simpleFileHandler
    - consoleHandler
```

You will then see the output written to junitxml (that's one of the things I planned on trying later today).

Reply:
It doesn't work but I didn't expect since all logs are captured from art.flow logger which is not propagating the logs to root logger, thus don't see why this change would affect it.

Maybe is would be better for you to provide me with a working minimal example where logs  appear in console and the same logs appear in the xmlfile?
Maybe it would be nice test addition to the existing ones as well.
Again, thanks for the help.

Reply:
I will get to this later when I get home and let you know.

Reply:
@KKoukiou out of the box your requirements (showing captured log in the terminal while tests are running *and* capture it so it appears in the junit xml) are not possible, I'm afraid.

But I managed to make things work like you expect by writing a fixture which manually adds the captured log information to the item's sections:

```python
@pytest.fixture(autouse=True)
def append_captured_log_to_item_stdout(request, caplog):
    yield
    for when in ('setup', 'call', 'teardown'):
        records = caplog.get_handler(when).records
        for record in records:
            request.node.add_report_section(when, 'stdout', record.message + '\n')
```

With this fixture in the root `conftest.py` file, I get the live logging output:

```
test_template.py::test_function_1
--------------------------------------------------- live log setup ----------------------------------------------------
test_template.py            30 INFO     Hi with info from test 1 setup
---------------------------------------------------- live log call ----------------------------------------------------
test_template.py            35 INFO     Hi with info from test 1
test_template.py            36 ERROR    Hi with error from test 1
PASSED                                                                                                           [100%]
```

And the log messages also appear in the xml file:

```xml
<?xml version="1.0" encoding="utf-8"?><testsuite errors="0" failures="0" name="pytest" skips="0" tests="1" time="0.079"><testcase classname="test_template" file="test_template.py" line="31" name="test_function_1" time="0.011009693145751953"><system-out>Hi with info from test 1 setup
Hi with info from test 1
Hi with error from test 1
</system-out></testcase></testsuite>
```

(This is obtained with your original logging configuration, except for the `propagate: 0` configuration which needs to be removed as commented before).

IMHO this is an acceptable workaround. What do you think?

Reply:
You 're super! I can definitely WA my use case! Thanks :)

Reply:
@KKoukiou can you try the following patch and test it (without the conftest.py fixture)?

```diff
diff --git a/_pytest/junitxml.py b/_pytest/junitxml.py
index 7fb40dc..0cbec7a 100644
--- a/_pytest/junitxml.py
+++ b/_pytest/junitxml.py
@@ -125,11 +125,14 @@ class _NodeReporter(object):
         self.append(node)
 
     def write_captured_output(self, report):
-        for capname in ('out', 'err'):
-            content = getattr(report, 'capstd' + capname)
-            if content:
-                tag = getattr(Junit, 'system-' + capname)
-                self.append(tag(bin_xml_escape(content)))
+        content = report.caplog   # TODO merge with report.capstdout or add a junit flag to switch between stdout and caplog
+        if content:
+            tag = getattr(Junit, 'system-out')
+            self.append(tag(bin_xml_escape(content)))
+        content = report.capstderr
+        if content:
+            tag = getattr(Junit, 'system-err')
+            self.append(tag(bin_xml_escape(content)))
 
     def append_pass(self, report):
         self.add_stats('passed')
diff --git a/_pytest/runner.py b/_pytest/runner.py
index 13abee3..b475be9 100644
--- a/_pytest/runner.py
+++ b/_pytest/runner.py
@@ -257,6 +257,14 @@ class BaseReport(object):
         return exc.strip()
 
     @property
+    def caplog(self):
+        """Return captured log lines, if log capturing is enabled
+
+        .. versionadded:: 3.4
+        """
+        return '\n'.join(content for (prefix, content) in self.get_sections('Captured log'))
+
+    @property
     def capstdout(self):
         """Return captured text from stdout, if capturing is enabled
```

@nicoddemus Do you have interest in integrating a fix for junitxml into pytest?

Reply:
@thisch @nicoddemus I confirm that with this patch I get the expected output in console and junitxml file as with the conftest.py adjustment.

Reply:
@thisch I'm not sure, TBH I would like to avoid to add more to this PR, it is quite big as it is already. Also I think merging capout and caplog needs more detailed discussion.

Reply:
Thx for the confirmation @KKoukiou. I totally agree that this PR is already quite large @nicoddemus. Let's create a new ticket for this issue.